### ISSUE TITLE
question about Input Size

### ISSUE THREAD

**Issue by xiaochengfuhuo** [jvanvugt/pytorch-unet#9]
Hi, thanks for showing your idea about structure and parameter.i have acquire a lot form it.
And i have a question about topic Input Size.
if we do not use padding , we will get a small size output . in your advice, you choose padding the output or cropping the lable_img to solve this problem. but why not resize the output or lable_img.
there may be some problem ,especially getting loss in trian, if using your method.
looking forward your reply.

**Comment by jvanvugt** [comment]
There are 3 problems with resizing the input or output tensor:
1. Still it is not possible to achieve every possible input size (e.g. you can't resize your input such that you get an odd output size)
2. Resizing causes aliasing effects, i.e., you might lose some information.
3. Convolutions are sensitive to the scale of the input. This might not be a problem in datasets of natural images for instance, because an object can appear at any distance from the camera, (and thus at different scales). However, in medical imaging for instance, there is often a physical meaning to the size of an object (e.g., 1 pixel is 0.1cm^2), in which case you probably don't want to resize images, because you make it harder for the model to learn.

In the case of U-Net, you also just can't resize the input or output, because that will mess up the skip-connections, since they won't align with the desired output

**Comment by xiaochengfuhuo** [comment]
thank for your reply. maybe i do not express my idea clearly. 
i mean that just resize the lable_img or last output when we want to get the loss. 
after read your reply, i am lost in thought .  i got that resize the last output may affect the gradient descent , because this process(resize) is not easy to get the gradient.  But resize the lable_img may not affect the train process(like the skip-connections). 

if we resize img, we must lose some information, but crop also lost the edge imformation. And is it possible that the model has not learn the  edge imformation if using cropping or padding.(e.g.,our task is img segmentation, after train ,we get a model, but this model only predict a part of the img(lose the edge part) ), i do not konw this situation will or not happen, it is just my conjecture.

and resize will change the physical meaning of the pixel,  it may do not matter to the segmentation task.because there are some paper use deformation to augment the data.

these are my thought of this problem, Welcome to correct my mistakes.

**Comment by jvanvugt** [comment]
In a lot of tasks it can indeed make sense to apply some kind of transformation to the image as data augmentation. Just naively resizing the label is a bit dangerous though, because remember that the U-net has skip connections to transfer lower-level features to the deeper layers. When you resize the label, these features do not necesserily get concatenated to the relevant location anymore, causing a drop in performance.

In general, padding the input with zeros should be safe without loss of accuracy. If you are concerned about border effects, you could even do some kind of reflection padding to reduce these issues.

As an aside, computing the gradient of resizing operations is generally quite easy.

**Comment by xiaochengfuhuo** [comment]
thanks a lot , it is very useful for me.
